### PR TITLE
[FIX] Delete removed user's subscriptions

### PR DESCRIPTION
**Pull request by Hudell**
Ref #6535 

This PR adds a migration to delete the subscriptions and empty rooms of users who have been removed.
It also changes the user deletion script to remove DM rooms and subscriptions to empty private rooms, using the same logic from the migration.

### CODE REVIEW

**Comment by ggazzo** [comment]
@Hudell update the migration please

File: packages/rocketchat-lib/server/functions/deleteUser.js [hunk 1]
```diff
@@ -19,12 +19,13 @@ RocketChat.deleteUser = function(userId) {
 		RocketChat.models.Subscriptions.db.findByUserId(userId).forEach((subscription) => {
```
Comment (ggazzo) [inline packages/rocketchat-lib/server/functions/deleteUser.js]: I think this code is wrong, we dont delete orphan subscriptions and we dont test if the user is the last owner/moderator and etc....

Comment (ggazzo) [reply packages/rocketchat-lib/server/functions/deleteUser.js]: RocketChat.models.Subscriptions.db.findByUserId(userId).forEach(({rid}) => {
            const room = RocketChat.models.Rooms.findOneById(rid);
            RocketChat.models.Subscriptions.removeByRoomId(rid);
			if (room) {
                if (room.t === 'd') {
                    RocketChat.models.Messages.removeByRoomId(rid);
					return RocketChat.models.Rooms.removeById(rid);
                }
                if (room.usernames && room.usernames.length === 1) {
                  RocketChat.models.Rooms.removeById(rid); // Remove non-channel rooms with only 1 user (the one being deleted)
                }
			}
		});

**Comment by rodrigok** [comment]
@Hudell can you update and fix the conflicts?

File: server/startup/migrations/v131.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+		subscriptions.forEach((subscription) => {
+			const room = RocketChat.models.Rooms.findOneById(subscription.rid);
+			if (room) {
+				// If the room is a direct message, remove all subscriptions and messages that it has
+				if (room.t === 'd') {
+					RocketChat.models.Subscriptions.removeByRoomId(subscription.rid);
+					RocketChat.models.Messages.removeByRoomId(subscription.rid);
+					RocketChat.models.Rooms.removeById(subscription.rid);
+				} else if (room.t !== 'c' && room.usernames.length === 1) {
```
Comment (rodrigok) [inline server/startup/migrations/v131.js]: 1. `room.usernames` isn't valid anymore.
2. Don't we delete the messages?

**Pull request merged**